main.go

```diff
@@ -393,7 +408,7 @@ func apiGroupForType(t *types.Type, typePkgMap map[*types.Type]*apiPackage) stri
 
 // anchorIDForLocalType returns the #anchor string for the local type
 func anchorIDForLocalType(t *types.Type, typePkgMap map[*types.Type]*apiPackage) string {
-	return fmt.Sprintf("%s.%s", apiGroupForType(t, typePkgMap), t.Name.Name)
+	return sanitizeId(fmt.Sprintf("%s.%s", apiGroupForType(t, typePkgMap), t.Name.Name))
```

Comment:
this is a breaking change as I understand it, hence not quite desitable

Reply:
I wasn't sure if you'd consider this a breaking change: I have no problem changing it.  I see three options:

- introduce a `sanitizeId` function called from templates, similar to the `asciiDocAttributeEscape` method I introduced. This is certainly the least complex and most flexible, but increases the API exposed to templates.
- sanitize ids based on the `asciiDoc` flag. This will probably end up with the most complex code, and will be the least flexible: perhaps users wanting html generation will want to have simplified IDs.
- introduce a `sanitizeId` flag.

I lean towards the first solution, exposing a `sanitizeId` function to templates.  What is your preference?

Reply:
I went ahead and pushed a commit to expose a `sanitizeId` method.  I had to add a `tryDereference` to `anchorIDForLocalType`: I think exposing this to templates without the dereference was a bug.

Travis was using Go 1.11: I updated to 1.12 to allow use of strings.ReplaceAll.  I'm not sure if there are other more modern API usages.  Is updating the Go version OK?